python/jupyterlab-chat/src/index.ts

```diff
@@ -791,13 +799,32 @@ const selectionWatcher: JupyterFrontEndPlugin<ISelectionWatcher> = {
   }
 };
 
+/**
+ * Extension providing the input toolbar registry.
+ */
+const inputToolbarFactory: JupyterFrontEndPlugin<IInputToolbarRegistryFactory> =
+  {
+    id: pluginIds.inputToolbarFactory,
+    description: 'The input toolbar registry plugin.',
+    autoStart: true,
+    provides: IInputToolbarRegistryFactory,
+    activate: (app: JupyterFrontEnd): IInputToolbarRegistryFactory => {
+      return {
+        create() {
+          return InputToolbarRegistry.defaultToolbarRegistry();
```

Comment:
Not sure how much work it would be, but could the JupyterLab `IToolbarWidgetRegistry` be reused?

In the case of the send / stop buttons, maybe instead of two distinct buttons it could be encapsulated as one widget, which renders differently based on some condition?

Reply:
> Not sure how much work it would be, but could the JupyterLab IToolbarWidgetRegistry be reused?

Thanks for the comment, I'll take a look at it, it may be worth using indeed.

> In the case of the send / stop buttons, maybe instead of two distinct buttons it could be encapsulated as one widget, which renders differently based on some condition?

In the example above, there's no others buttons, but if there were, I suppose we'd want to hide them too. For example an *attachments* button, that would not be necessary during the bot response.

Reply:
As far as I understand, `IToolbarWidgetRegistry` is designed to work with lumino `Widget`. The chat widget is full react (embedded in a `ReactWidget`), so I don't really know how it can be compatible without a big refactoring of the chat itself.

Reply:
ok, then let's think about that later. Thanks for checking!